storage/sqlStorages/template.go

```diff
@@ -0,0 +1,175 @@
+package sqlStorages
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/cube2222/octosql"
+	"github.com/cube2222/octosql/config"
+	"github.com/cube2222/octosql/execution"
+	"github.com/cube2222/octosql/physical"
+	"github.com/cube2222/octosql/physical/metadata"
+	"github.com/pkg/errors"
+)
+
+type SQLSourceTemplate interface {
+	GetIPAddress(dbConfig map[string]interface{}) (string, int, error)
+	GetConnection(user, password, host, dbName string, port int) (*sql.DB, error)
```

Comment:
You could change this to GetDSN, and just return the string you use on db.open

Reply:
I could, but that would require a `GetDriverName` function, so that MySQL template can pass `mysql` as the driver name, so I think it's better this way

Reply:
You can do both, and this way abstract away the connecting part :) Will be easier to test too.

Reply:
Just return two values

Reply:
Or create a struct ConnectionInfo